File: .sdk/tm/rb/utility/feature_hook.rb
# N404ErrorHandler SDK utility: feature_hook
module N404ErrorHandlerUtilities
  FeatureHook = ->(ctx, name) {
    return unless ctx.client
    features = ctx.client.features
    return unless features
    features.each do |f|
      f.send(name, ctx) if f.respond_to?(name)
    end
  }
end

File: .sdk/tm/php/utility/FeatureHook.php
<?php
declare(strict_types=1);

// N404ErrorHandler SDK utility: feature_hook

class N404ErrorHandlerFeatureHook
{
    public static function call(N404ErrorHandlerContext $ctx, string $name): void
    {
        if (!$ctx->client) {
            return;
        }
        $features = $ctx->client->features ?? null;
        if (!$features) {
            return;
        }
        foreach ($features as $f) {
            if (method_exists($f, $name)) {
                $f->$name($ctx);
            }
        }
    }
}
